packages/reports/addon/components/report-builder/sidebar.ts

```diff
@@ -120,14 +120,36 @@ export default class ReportBuilderSidebar extends Component<Args> {
   }
 
   @task *fetchDataSources(): TaskGenerator<SourceItem<NaviDataSource>[]> {
-    const sources: SourceItem<NaviDataSource>[] = sortBy(config.navi.dataSources, ['displayName']).map(
-      (dataSource) => ({
+    let sources: SourceItem<NaviDataSource>[] = [];
+    const fetchNamespacesTasks: TaskInstance<SourceItem<NaviDataSource>[]>[] = [];
+
+    config.navi.dataSources.forEach((dataSource) => {
+      sources.push({
         name: dataSource.displayName,
         description: dataSource.description,
         source: dataSource,
-      })
-    );
-    return yield sources;
+      });
+
+      fetchNamespacesTasks.push(taskFor(this.fetchDataSourceNamespaces).perform(dataSource));
+    });
+
+    const namespaces = (yield all(fetchNamespacesTasks)).flat();
+    sources = [...sources, ...namespaces];
+    return yield sortBy(sources, ['name']);
+  }
+
+  @task *fetchDataSourceNamespaces(dataSource: NaviDataSource): TaskGenerator<SourceItem<NaviDataSource>[]> {
```

Comment:
Long term it would be good to have this layer not know about namespaces. We can just process the original data sources with nested namespaces and generate a single flat list of `data sources`.

Reply:
I agree, like you and @kevinhinterlong suggested we should maybe have a service around that to discover namespaces and flatten